lib/entity_manager.js

```diff
@@ -55,9 +60,12 @@ EntityManager.prototype.query = function() {
   }
 
   var result = []
-  for (var id = 0; id < this._entityCounter; id++) {
-    if (this._entityFlag[id] === ENTITY_ALIVE && (this._entityMask[id] & mask) === mask) {
-      result.push(this._entityInst[id])
+
+  var entityMasks = Object.keys(this._entityCache)
```

Comment:
I tried two approaches so far, changing the property descriptor to enumerable: false so it doesn't appear in Object.keys(). This resulted in quite bad performance when adding/removing components.

Also tried simply deleting the property which I know is a bad thing to do on objects. But the performance is much better. 10x faster when adding components, 100x faster when removing them.

Any other ideas on how to do this?